eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/OperationMethodProviders/ResourceOperationMethodProvider.cs

```diff
@@ -38,21 +41,39 @@ internal class ResourceOperationMethodProvider
         private readonly bool _isLongRunningOperation;
         private readonly bool _isFakeLongRunningOperation;
 
+        /// <summary>
+        /// Creates a new instance of <see cref="ResourceOperationMethodProvider"/> which represents a method on a client
+        /// </summary>
+        /// <param name="provider">The carrier of this operation. </param>
+        /// <param name="resource">The corresponding resource of this operation. </param>
+        /// <param name="restClient">The client provider for this operation to send the request. </param>
+        /// <param name="method">The input service method that we are building from. </param>
+        /// <param name="convenienceMethod">The corresponding convenience method provided by the generator framework. </param>
+        /// <param name="clientDiagnosticsField">The field that holds the client diagnostics instance. </param>
+        /// <param name="restClientField">The field that holds the rest client instance. </param>
+        /// <param name="isAsync">Whether this method is an async method. </param>
         public ResourceOperationMethodProvider(
-            ResourceClientProvider resourceClientProvider,
+            ContextualClientProvider provider,
+            ResourceClientProvider resource,
```

Comment:
The first one is passed in as ResourceCollectionClientProvider. If we remove the inheritance of ContextualClientProvider from ResourceClientProvider, we would need to pass ContextualClientProvider to PopulateArguments and TryGetContextualParameter.
And this is getting confusing to mix resourceCollection and resource in this MethodProvider, might consider to split them as well.

Reply:
We must need the `ResourceClientProvider`.
We need it for checking if we should wrap ResourceData into resource.
If we remove that inheritance, it will only end with we need one more parameter: the enclosing type of this operation, and a thing providing the contextual parameters.